packages/run-protocol/src/metrics.js

```diff
@@ -0,0 +1,40 @@
+// @ts-check
+import { makeStoredSubscription, makeSubscriptionKit } from '@agoric/notifier';
+import { E } from '@endo/eventual-send';
+
+const PATH = 'metrics';
+
+/**
+ * @template T
+ * @param {ERef<StorageNode>} [storageNode]
+ * @param {ERef<Marshaller>} [marshaller]
+ * @param {T[]} optionalInitialState
+ * @returns {MetricsSubscriptionKit<T>}
+ */
+export const makeMetricsSubscriptionKit = (
+  storageNode,
+  marshaller,
+  ...optionalInitialState
+) => {
+  const {
+    publication: metricsPublication,
+    subscription: rawMetricsSubscription,
+  } = makeSubscriptionKit(...optionalInitialState);
+
+  const metricsStorageNode = storageNode && E(storageNode).getChildNode(PATH);
+  const metricsSubscription = makeStoredSubscription(
+    rawMetricsSubscription,
+    metricsStorageNode,
+    marshaller,
+  );
+
+  return { metricsPublication, metricsSubscription };
```

Comment:
It's a great helper, and we can make it more general without baking "metrics" in.  Then it can be part of the notifier package, too.

```suggestion
  return { publication, subscription };
```